### PR TITLE
Confirmation box for borers overdosing their hosts with chemicals

### PR DESCRIPTION
**Pull request by 9600bauds**
Now it's just a confirmation box
Also changes the default logging color to blue so you don't get a red message every time a borer secretes anything, does give you a bright red message if a borer ticks "yes" to the OD confirmation

~~As per discussion in http://ss13.moe/forum/viewtopic.php?f=27&t=2368&start=40 and this strawpoll: http://strawpoll.me/7403344/r (15 votes to 'killing a host is bad' and 1 to killing a host is OK' at the time of this PR)~~

~~I think limiting what a borer can do in the rules wouldn't be quite the best choice, because a borer should still be able to hold a grudge against their host's murderer and there's nothing wrong with that. This aims to simply remove the "jump into guy you grudge just to OD them" tactic.~~

~~If you don't like your host, get a new one.~~

### CODE REVIEW

**Comment by chatsonf** [comment]
Borers can become antags, which would gimp them from doing anything
-1.
Also it is still being discussed currently.

**Comment by 9600bauds** [comment]
> Borers can become antags

How? Since when?

**Comment by chatsonf** [comment]
Autotraitor can select them to be a traitor.
please don't remove it, IT'S A FEATUREEEE

**Comment by 9600bauds** [comment]
Could easily be arranged to let autotatored borers still overdose their host, but then we're getting kinda meta-ey

Besides, jumping on someone else and trying to get them to kill your target seems way more fun for everyone involved

**Comment by chatsonf** [comment]
You still need to be able to convince them to kill others, and that won't work unless you have a way to threaten them

**Comment by PJB3005** [comment]
But how will I post round grief as borer now?

**Comment by Intigracy** [comment]
Tbh I like this change simply because of the amount of accidents that already happen by borers overdosing, not even considering the intentional overdoses.

**Comment by clusterfack** [comment]
Perhaps a confirmation box instead

**Comment by Probe1** [comment]
What @clusterfack said.  The problem stemmed from me telling a player something was fine and it wasn't.  The solution is for me to not say it's fine.  We don't need a coding solution.

If you remove it, what happens in the rare but has happened situation where a borer gets auto-traitored?  Why code around it when borers poisoning their hosts has only come up like, a half dozen times since they were made non antags?

BUT WHILE YOU'RE AT IT REMOVE THEM FROM THE FUCKIN LIST OF ANTAGS WHEN COMEONE CHECKS ANTAG.

**Comment by jknpj** [comment]
Overdosing an antag that is killing the whole crew or other borer's host is the best way to make sure your "children" will have host to infest AND that you will have backup hosts. It IS about the survival of your species in the end, but it is also VERY abusable.

**Comment by 9600bauds** [comment]
Problem here is that overdosing is fairly shitty because it's one-sided, you can't even kick the borer out of your head, you need to get surgery to get it out. So if you're any type of antag, you're probably better off not getting a borer at all. And this is a can of worms in terms of problems, because then you're shitting it up for any borers that would actually want to infest an antagonist. Other people might even be suspicious of you for not wanting a borer.

I have absolutely zero problems with borers not liking antagonists, I just think the overdosing tactic is just worse off for everyone involved.

**Comment by PJB3005** [comment]
Shouldn't have let the borer inside your head in the first place then.

**Comment by chatsonf** [comment]
What if you get knocked down by ZAS? A borer who is allowed to kill is on the levels of "EI NATH", there isn't counterplay to it

**Comment by ihadtoregisterforthis** [comment]
Borers are told to cooperate with their host:

> Being as helpful to your host as possible is the best option for survival

Borers shouldn't be allowed to kill hosts, because that could open situations of metaknowledge and killing from the grave. But blocking them with codewalls doesn't sound right.
How about adding the OD levels to the reagents list to minimize accidents instead?

Besides, would you rather help the apex predator to gain their love or plop out to be predated with the victims?

**Comment by 9600bauds** [comment]
> Shouldn't have let the borer inside your head in the first place then. 

PJB I believe most of my comment was addressing that

> So if you're any type of antag, you're probably better off not getting a borer at all. And this is a can of worms in terms of problems, because then you're shitting it up for any borers that would actually want to infest an antagonist. Other people might even be suspicious of you for not wanting a borer.

**Comment by 9600bauds** [comment]
I think the best solution here would be adding something in the rules about not overdosing anyone, but allowing you to try to get another host to kill the murderer etc.

And I'll probably make this into a confirmation box rather than a codewall, yeah

**Comment by kol1th** [comment]
> I think the best solution here would be adding something in the rules about not overdosing anyone, but allowing you to try to get another host to kill the murderer etc.

I feel like that's the best solution, as well.

**Comment by Intigracy** [comment]
Confirmation box would still be nice, because accidents still happen.

**Comment by 9600bauds** [comment]
Now it's just a confirmation box

**Comment by 9600bauds** [comment]
I forgot to remove them from checkantags

**Comment by chatsonf** [comment]
IT'S A FEATUREEEE

**Comment by 9600bauds** [comment]
there

**Comment by Probe1** [comment]
nice ATOMIC pr

**Comment by 9600bauds** [comment]
thanks bb :rotating_light:

File: code/modules/mob/living/simple_animal/borer.dm [hunk 1]
```diff
@@ -447,12 +447,17 @@ var/global/borer_unlock_types = typesof(/datum/unlockable/borer) - /datum/unlock
...
+		if(alert("Secreting that much [chemID] would cause an overdose in your host. Are you sure?", "Secrete Chemicals", "Yes", "No") != "Yes")
+			return
+		add_gamelogs(src, "intentionally overdosed \the [host] with '[chemID]'", admin = TRUE, tp_link = TRUE, span_class = "danger")
 
 	if(!host || controlling || !src || stat) //Sanity check.
 		return
 
 	to_chat(src, "<span class='info'>You squirt a measure of [chem.name] from your reservoirs into [host]'s bloodstream.</span>")
-	add_gamelogs(src, "secreted [units]U of '[chemID]' into \the [host]", admin = TRUE, tp_link = TRUE, span_class = "danger")
+	add_gamelogs(src, "secreted [units]U of '[chemID]' into \the [host]", admin = TRUE, tp_link = TRUE, span_class = "notice")
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/borer.dm]: Can you change this to the adminlog span class

Comment (9600bauds) [inline code/modules/mob/living/simple_animal/borer.dm]: uh
which one is that

Comment (clusterfack) [inline code/modules/mob/living/simple_animal/borer.dm]: The very light blue one that adminsay uses

Comment (9600bauds) [inline code/modules/mob/living/simple_animal/borer.dm]: it is done as thy asked

**Pull request merged**